src/simplify_qdq.cpp

```diff
@@ -0,0 +1,205 @@
+                match::arg(1)(match::name("dequantizelinear")(
+                    match::arg(0)(match::name(get_all_op_names()).bind("q1")).bind("dq1"))))
+                .bind("qop")));
+    }
+
+    void apply(module& m, match::matcher_result r) const
+    {
+        auto op  = r.result;
+        auto qop = r.instructions["qop"];
+        auto dq0 = r.instructions["dq0"];
+        auto q0  = r.instructions["q0"];
+        auto dq1 = r.instructions["dq1"];
+        auto q1  = r.instructions["q1"];
+
+        // Only INT8 type currently supported
+        if(q0->get_shape().type() != migraphx::shape::int8_type or
+           q1->get_shape().type() != migraphx::shape::int8_type)
+            return;
+
+        // Only zero_point==0 currently supported
+        auto dq0_args  = dq0->inputs();
+        auto dq1_args  = dq1->inputs();
+        bool all_zeros = true;
+        all_zeros &= inputs_are_zeros(dq0_args.at(2));
+        all_zeros &= inputs_are_zeros(dq1_args.at(2));
+        if(not all_zeros)
+            return;
+
+        auto scale     = get_scale(dq0_args.at(1)) * get_scale(dq1_args.at(1));
+        auto qop_args  = qop->inputs();
+        qop_args.at(0) = q0;
+        qop_args.at(1) = q1;
+        instruction_ref dq;
+        instruction_ref dq_scale;
+        instruction_ref zero_point;
+        if(qop->name() == "convolution")
+        {
+            auto conv_op = any_cast<op::convolution>(qop->get_operator());
+            qop          = m.insert_instruction(qop,
+                                       migraphx::make_op("quant_convolution",
+                                                         {{"padding", conv_op.padding},
+                                                          {"stride", conv_op.stride},
+                                                          {"dilation", conv_op.dilation},
+                                                          {"padding_mode", conv_op.padding_mode},
+                                                          {"group", conv_op.group}}),
+                                       qop_args);
+            dq_scale     = m.add_literal(static_cast<float>(scale));
+            zero_point   = m.add_literal(0);
+        }
+        else if(qop->name() == "dot")
+        {
+            auto dot_op    = any_cast<op::dot>(qop->get_operator());
+            auto scale_val = dot_op.alpha / scale;
+            auto alpha     = 1;
+            auto beta      = (qop->inputs().size() == 3) ? dot_op.beta : 0.0f;
+            qop            = m.insert_instruction(
+                qop, migraphx::make_op("quant_dot", {{"alpha", alpha}, {"beta", beta}}), qop_args);
+            dq_scale   = m.add_literal(static_cast<float>(scale_val));
+            zero_point = m.add_literal(static_cast<int>((-1.0f * beta) / scale_val));
+        }
+        else
+        {
+            MIGRAPHX_THROW(qop->name() + " does not currently have a quantized implementation.");
```

Comment:
We shouldn't throw an exception. We should only apply the transformation for dot and convolution.